### PR TITLE
Copter: convert to use AC_AutoTune library

### PR DESCRIPTION
**Pull request by tridge**
this maintains existing behavior

### CODE REVIEW

File: ArduCopter/mode.h [hunk 1]
```diff
@@ -439,157 +439,44 @@ class ModeAuto : public Mode {
...
+{
+public:
+    bool init() override;
+    void run() override;
+
+protected:
+    bool start(void) override;
+    bool position_ok() override;
+    float get_pilot_desired_climb_rate_cms(void) const override;
+    void get_pilot_desired_rp_yrate_cd(int32_t &roll_cd, int32_t &pitch_cd, int32_t &yaw_rate_cds) override;
```
Comment (rmackay9) [inline ArduCopter/mode.h]: a little typo in the function name I guess?  it should be get_pilot_desired_rpy_rate_cd probably?

Comment (tridge) [reply ArduCopter/mode.h]: no, r and p are attitudes, y is a yaw rate
what name do you suggest?

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
One small thing is that to be consistent with how we do other overrides of shared classes is AC_AutoTune should be moved to a new AC_AutoTune.h and AC_AutoTune.cpp file within the ArduCopter directory.

**Pull request merged**